docs/concepts/registries.md

```diff
@@ -28,32 +28,29 @@ public ExampleMod() {
 }
 ```
 
-### `Register` events
+### `RegisterEvent`
 
-The `RegistryEvent`s are the second and more flexible way to register objects. These [events][] are fired after the mod constructors and before the loading of configs.
-
-The event used to register objects is `RegistryEvent$Register<T>`. The type parameter `T` should be set to the type of the object being registered. Calling `#getRegistry` will return the registry, upon which objects are registered with `#register` or `#registerAll`. 
+`RegisterEvent` is the second way to register objects. This [event] is fired for each registry after the mod constructors and before the loading of configs. Objects are registered using `#register` by passing in the registry key, the name of the registry object, and the object itself. There is an additional `#register` overload which takes in a consumed helper to register an object with a given name. It is recommended to use this method to avoid unnecessary object creation.
 
 Here is an example: (the event handler is registered on the *mod event bus*)
 
 ```java
 @SubscribeEvent
-public void registerBlocks(RegistryEvent.Register<Block> event) {
-  event.getRegistry().registerAll(new Block(...), new Block(...), ...);
+public void register(RegisterEvent event) {
+  event.register(ForgeRegistries.Keys.BLOCKS,
+    helper -> helper.register(new ResourceLocation(MODID, "rock"), new Block(...))
+  );
 }
 ```
 
 ### Registries that aren't Forge Registries
 
-Due to some peculiarities of vanilla code, not all registries are wrapped by Forge. These can be static registries, like `RecipeType`, which are safe to use. There are also dynamic registries, like `ConfiguredFeature` and some other worldgen registries, which are typically represented in JSON. These objects should only be registered this way if there is another registry object that requires it. `DeferredRegister#create` has an overload which allows modders to specify the registry key of which vanilla registry to create a `RegistryObject` for. The registry method and attaching to the mod event bus is the same as other `DeferredRegister`s.
+Not all registries are wrapped by Forge. These can be static registries, like `LootItemConditionType`, which are safe to use. There are also dynamic registries, like `ConfiguredFeature` and some other worldgen registries, which are typically represented in JSON. These objects should only be registered this way if there is another registry object that requires it. `DeferredRegister#create` has an overload which allows modders to specify the registry key of which vanilla registry to create a `RegistryObject` for. The registry method and attaching to the mod event bus is the same as other `DeferredRegister`s.
```

Comment:
For "These objects should only be registered this way if there is another registry object that requires it.", the "objects" referenced here should be clarified as being objects from the dynamic registries.

Also, perhaps an admonition would be more suited for that?